migration/migration1.go

```diff
@@ -0,0 +1,175 @@
+package migration
+
+import (
+	"encoding/json"
+	"fmt"
+	"sync"
+
+	"github.com/ipfs/go-cid"
+	"github.com/ipfs/go-datastore"
+	"github.com/ipfs/go-datastore/query"
+	"github.com/textileio/powergate/ffs"
+	"github.com/textileio/powergate/util"
+)
+
+// V1MultitenancyMigration contains the logic to upgrade a datastore from
+// version 0 to version 1. Transactionality is disabled since is a big migration.
+var V1MultitenancyMigration = Migration{
+	UseTxn: false,
+	Run: func(ds datastoreReaderWriter) error {
+		cidOwners, err := v0CidOwners(ds)
+		if err != nil {
+			return fmt.Errorf("getting cid owners: %s", err)
+		}
+		log.Infof("Starting job logger migration...")
+		if err := migrateJobLogger(ds, cidOwners); err != nil {
+			return fmt.Errorf("migrating job logger: %s", err)
+		}
+		log.Infof("Job logger migration finished")
+
+		log.Infof("Starting storage info migration...")
+		if err := migrateStorageInfo(ds, cidOwners); err != nil {
+			return fmt.Errorf("migrating storage info: %s", err)
+		}
+		log.Infof("Storage info migration finished")
+
+		log.Infof("Starting trackstore migration...")
+		if err := migrateTrackstore(ds, cidOwners); err != nil {
+			return fmt.Errorf("migrating trackstore: %s", err)
+		}
+		log.Infof("Trackstore migration finished")
+
+		log.Infof("Starting pinstore filling migration...")
+		if err := pinstoreFilling(ds, cidOwners); err != nil {
+			return fmt.Errorf("filling pinstore: %s", err)
+		}
+		log.Infof("Pinstore filling migration finished")
+
+		return nil
+	},
+}
+
+func v0CidOwners(ds datastoreReaderWriter) (map[cid.Cid][]ffs.APIID, error) {
+	iids, err := v0APIIDs(ds)
+	if err != nil {
+		return nil, fmt.Errorf("getting v0 iids: %s", err)
+	}
+
+	lim := make(chan struct{}, 1000)
+	var lock sync.Mutex
+	var errors []string
+	owners := map[cid.Cid][]ffs.APIID{}
+	for _, iid := range iids {
+		lim <- struct{}{}
+		iid := iid
+		go func() {
+			defer func() { <-lim }()
+			cids, err := v0GetCidsFromIID(ds, iid)
+			if err != nil {
+				lock.Lock()
+				errors = append(errors, fmt.Sprintf("getting cids from iid: %s", err))
+				lock.Unlock()
+				return
+			}
+			lock.Lock()
+			for _, c := range cids {
+				owners[c] = append(owners[c], iid)
+			}
+			lock.Unlock()
+		}()
+	}
+
+	for i := 0; i < len(lim); i++ {
+		lim <- struct{}{}
+	}
+
+	if len(errors) > 0 {
+		for _, m := range errors {
+			log.Error(m)
+		}
+		return nil, fmt.Errorf("building cidowners had %d errors", len(errors))
+	}
+
+	return owners, nil
+}
+
```

Comment:
All below `v0` functions are helpers for the migration stuff, which resolves some important questions for migrations:
- Which iids exists.
- Which iids own a Cid
- Get the storage config of a Cid for a iid.

These methods should be agnostic of internal components, since if that is the case we could handcuff improving internal components in the future since migrations rely on them.